detectable/src/test/java/com/synopsys/integration/detectable/detectables/xcode/PackageResolvedTransformerTest.java

```diff
@@ -0,0 +1,110 @@
+package com.synopsys.integration.detectable.detectables.xcode;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.Test;
+
+import com.synopsys.integration.bdio.graph.DependencyGraph;
+import com.synopsys.integration.bdio.model.Forge;
+import com.synopsys.integration.bdio.model.externalid.ExternalId;
+import com.synopsys.integration.bdio.model.externalid.ExternalIdFactory;
+import com.synopsys.integration.detectable.detectables.xcode.model.PackageResolved;
+import com.synopsys.integration.detectable.detectables.xcode.model.PackageState;
+import com.synopsys.integration.detectable.detectables.xcode.model.ResolvedObject;
+import com.synopsys.integration.detectable.detectables.xcode.model.ResolvedPackage;
+import com.synopsys.integration.detectable.util.graph.GraphAssert;
+
+class PackageResolvedTransformerTest {
+    private final ExternalIdFactory externalIdFactory = new ExternalIdFactory();
+    private final ExternalId swiftCollections = createId(externalIdFactory, "apple/swift-collections", "1.0.1");
+    private final ExternalId auth0 = createId(externalIdFactory, "auth0/Auth0.swift", "1.35.0");
+    private final ExternalId rSwiftLibrary = createId(externalIdFactory, "mac-cain13/R.swift.Library", "5.4.0");
+    private final ExternalId swiftLog = createId(externalIdFactory, "apple/swift-log", "1.4.2");
+    private final String defaultFileFormatVersion = PackageResolvedTransformer.KNOWN_FILE_FORMAT_VERSIONS[0];
+
+    @Test
+    void simpleTest() {
+        PackageResolvedTransformer transformer = new PackageResolvedTransformer(externalIdFactory);
+        PackageResolved packageResolved = createPackageResolved(null, defaultFileFormatVersion);
+
+        DependencyGraph dependencyGraph = transformer.transform(packageResolved);
+        assertDefaultPackagesExist(dependencyGraph);
+    }
+
+    @Test
+    void malformedUrlTest() {
```

Comment:
Why not just test the url parsing function? Seems like it's testing more than a unit.